### PR TITLE
Add metrics for digest cache

### PR DESCRIPTION
**Pull request by dmage**

### CODE REVIEW

File: pkg/dockerregistry/server/cache/cache_test.go [hunk 1]
```diff
@@ -104,25 +104,15 @@ func TestDigestCacheAddRepository(t *testing.T) {
...
-	for _, v := range repos[0:2] {
-		if item.repositories.Contains(v) {
-			t.Fatalf("%q found in the repositories", v)
-		}
-	}
-
-	if !item.repositories.Contains(repos[len(repos)-1]) {
-		t.Fatalf("%q not found in the repositories", repos[len(repos)-1])
+	dgstRepos := cache.Repositories(dgst)
+	if len(dgstRepos) != 1 && dgstRepos[0] != repos[len(repos)-1] {
```
Comment (bparees) [inline pkg/dockerregistry/server/cache/cache_test.go]: s/&&/||/ ?

**Comment by dmage** [comment]
[(8) Implement recommended registry metrics [registry]](https://trello.com/c/g3380oUO/1534-8-implement-recommended-registry-metrics-registry)

**Comment by dmage** [comment]
@bparees @legionus ptal

**Comment by dmage** [comment]
/retest
do it!

**Pull request merged**